h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,204 @@
+package hex.genmodel.tools;
+
+import com.mxgraph.layout.mxCompactTreeLayout;
+import com.mxgraph.layout.mxEdgeLabelLayout;
+import com.mxgraph.layout.mxIGraphLayout;
+import com.mxgraph.util.mxCellRenderer;
+import hex.genmodel.algos.tree.ConvertTreeOptions;
+import hex.genmodel.algos.tree.SharedTreeGraph;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.genmodel.algos.tree.TreeBackedMojoModel;
+import org.jgrapht.ext.JGraphXAdapter;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedMultigraph;
+import org.jgrapht.io.*;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    @Override
+    public void run() throws Exception {
+        if (!Format.png.equals(format)){
+            super.run();
+        } else {
+            validateArgs();
+            if (genModel instanceof TreeBackedMojoModel){
+                TreeBackedMojoModel treeBackedModel = (TreeBackedMojoModel) genModel;
+                ConvertTreeOptions options = new ConvertTreeOptions().withTreeConsistencyCheckEnabled();
+                final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+                printPng(g);
+            }
+            else {
+                System.out.println("ERROR: Unknown MOJO type");
+                System.exit(1);
+            }
+        } 
+    }
+
+    @Override
+    public boolean supportsFormat(Format format){
+        return true;
+    }
+    
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        Path outputDirectoryPath = Paths.get(outputFileName);
+        int numberOfTrees = trees.subgraphArray.size();
+        if (numberOfTrees > 1) { 
+            if (outputFileName == null) {
+                outputFileName = Paths.get("").toString();
+            }
+            if (Files.exists(outputDirectoryPath) && !Files.isDirectory(outputDirectoryPath)) {
+                Files.delete(outputDirectoryPath);
```

Comment:
we should never delete anything, overwriting things withtout asking is bad enough.
main reason: what if output dir would be current working directory? than this would fail, but why couldnt the output be the working dir?

Reply:
this will not delete directory (...&& !Files.isDirectory(outputDirectoryPath)). If we want to create directory of pictures of trees and there is already file named like the directory to be created then this file is deleted so that directory can be created - as an analogy to overwriting the old file with the new one. But can be redone to just fail with an error.

Reply:
got it, thanks